packages/widgets/src/blocks/legacy-widget/index.php

```diff
@@ -114,7 +114,7 @@ function handle_legacy_widget_preview_iframe() {
 	exit;
 }
 
-// Ensure handle_legacy_widget_preview_iframe() is called after Core's
-// register_block_core_legacy_widget() (priority = 10) and after Gutenberg's
-// register_block_core_legacy_widget() (priority = 20).
-add_action( 'init', 'handle_legacy_widget_preview_iframe', 21 );
+// Use admin_init instead of init to ensure get_current_screen function is already available.
+// This isn't strictly required, but enables better compatibility with existing plugins.
+// See: https://github.com/WordPress/gutenberg/issues/32624.
+add_action( 'admin_init', 'handle_legacy_widget_preview_iframe', 20 );
```

Comment:
Why `20`? It used to be `21` so that it would fire after Gutenberg registered blocks at priority `20`, but that's no longer relevant since `admin_init` always fires after `init`.

Would sticking to the default priority (`10`) work just as well?

```suggestion
add_action( 'admin_init', 'handle_legacy_widget_preview_iframe' );
```

Reply:
I think it would work just as well, I just wanted to list the priority explicitly without relying on the defaults. 10 would be fine too :-)